frontend/restInterfaces/BaseRestInterface.ts

```diff
@@ -0,0 +1,65 @@
+import axios from 'axios';
+
+/**
+ * Base Class responsible for the interaction with the server. Implements different types of HTTP-Methods.
+ */
+export abstract class BaseRestInterface {
+  /**
+   * Sends GET request to backend
+   * @param url the url
+   * @returns server response
+   * @protected
+   */
+  protected static async get<T>(url: string): Promise<T> {
+    const { data } = await axios.get<T>(url);
```

Comment:
The url needs to be concatinated with the base url (e.g. `http://localhost:5000`). This should be changeble over the vite enviroment.

Reply:
This accounts for all methods in this class

Reply:
We dont even need to make it changable via the vite environment, if we use something like this, which should automatically switch over between local environment and deployed one
`const backendHost = (window.location.hostname.includes('localhost') || window.location.hostname.includes('127.0.0.1')) ? 'http://localhost:5000' : window.location.origin;`